tests/ui_tools/test_popups.py

```diff
@@ -343,14 +355,66 @@ def test__fetch_user_data(
 
         mocker.patch.object(self.controller.model, "get_user_info", return_value=data)
 
-        display_data = self.user_info_view._fetch_user_data(self.controller, 1)
+        display_data, custom_profile_data = self.user_info_view._fetch_user_data(
+            self.controller, 1
+        )
 
         assert display_data.get(expected_key, None) == expected_value
 
+    @pytest.mark.parametrize(
+        [
+            "to_vary_in_each_user",
+            "expected_value",
+        ],
+        [
+            case(
+                [],
+                {},
+                id="user_has_no_custom_profile_data",
+            ),
+            case(
+                [
+                    {
+                        "label": "Biography",
+                        "value": "Simplicity",
+                        "type": 2,
+                        "order": 2,
+                    },
+                    {
+                        "label": "Mentor",
+                        "value": [11, 12],
+                        "type": 6,
+                        "order": 7,
+                    },
+                ],
+                {"Biography": "Simplicity", "Mentor": "Human 1, Human 2"},
+                id="user_has_custom_profile_data",
+            ),
+        ],
+    )
+    def test__fetch_user_data_custom_profile_data(
+        self,
+        mocker: MockerFixture,
+        to_vary_in_each_user: Dict[str, Any],
+        expected_value: Dict[str, str],
+    ) -> None:
+        data = dict(self.user_data)
+        data["custom_profile_data"] = to_vary_in_each_user
+
+        mocker.patch.object(self.controller.model, "get_user_info", return_value=data)
+
+        display_data, custom_profile_data = self.user_info_view._fetch_user_data(
+            self.controller, 1
+        )
+
+        assert custom_profile_data == expected_value
+
     def test__fetch_user_data_USER_NOT_FOUND(self, mocker: MockerFixture) -> None:
         mocker.patch.object(self.controller.model, "get_user_info", return_value=dict())
 
-        display_data = self.user_info_view._fetch_user_data(self.controller, 1)
+        display_data, custom_profile_data = self.user_info_view._fetch_user_data(
```

Comment:
assert on custom_profile_data? The other two tests account for each part individually, but this is the only test for invalid user.